src/openrct2/management/Research.cpp

```diff
@@ -591,28 +591,20 @@ void scenery_set_not_invented(const ScenerySelection& sceneryItem)
 
 bool scenery_group_is_invented(int32_t sgIndex)
 {
-    auto invented = false;
     const auto sgEntry = get_scenery_group_entry(sgIndex);
     if (sgEntry != nullptr && sgEntry->entry_count > 0)
```

Comment:
Is this check still useful now? I'd guess that this is made obsolete by the check `item.entryIndex == sgIndex` later on.

Reply:
I originally simplified it to just the none_of but then when you look at the caller its going to pass in completely invalid sgIndex's a lot of the time so might as well save some searching time by doing a quick nullptr check.

Reply:
I would prefer this returns early if its null, reduces a scope.